src/workerd/api/actor-state.h

```diff
@@ -232,7 +235,7 @@ class DurableObjectStorage: public jsg::Object, public DurableObjectStorageOpera
   }
 
 private:
-  IoPtr<ActorCache> cache;
+  kj::OneOf<IoPtr<ActorCache>, IoPtr<ActorSqlite>> cache;
```

Comment:
Yeah, the difficulty is that `deleteAll()` and `transaction()` cannot be implemented in terms of `ActorCacheInterface` alone, which is in turn because `ActorCacheInterface` is designed to be the superclass of `ActorCache` and `ActorCache::Transaction`. I guess I could make a second intermediate class `ActorCacheInterfaceWithDeleteAllAndTransaction` which is only inherited by `ActorCache` and `ActorSqlite` but not `ActorCache::Transaction`, but (1) I'd need to think of a better name, (2) the extra class hierarchy brings its own complexity, I'm not sure it would be better.